packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -131,57 +131,7 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 
 	return useMemo(
 		() => ( {
-			...Object.fromEntries(
-				Object.entries( settings ).filter( ( [ key ] ) =>
-					[
-						'__experimentalBlockDirectory',
-						'__experimentalDiscussionSettings',
-						'__experimentalFeatures',
-						'__experimentalPreferredStyleVariations',
-						'__experimentalSetIsInserterOpened',
-						'__unstableGalleryWithImageBlocks',
-						'alignWide',
-						'allowedBlockTypes',
-						'blockInspectorTabs',
-						'allowedMimeTypes',
-						'bodyPlaceholder',
-						'canLockBlocks',
-						'codeEditingEnabled',
-						'colors',
-						'disableCustomColors',
-						'disableCustomFontSizes',
-						'disableCustomSpacingSizes',
-						'disableCustomGradients',
-						'disableLayoutStyles',
-						'enableCustomLineHeight',
-						'enableCustomSpacing',
-						'enableCustomUnits',
-						'focusMode',
-						'fontSizes',
-						'gradients',
-						'generateAnchors',
-						'hasFixedToolbar',
-						'isDistractionFree',
-						'hasInlineToolbar',
-						'imageDefaultSize',
-						'imageDimensions',
-						'imageEditing',
-						'imageSizes',
-						'isRTL',
-						'keepCaretInsideBlock',
-						'maxWidth',
-						'onUpdateDefaultBlockStyles',
-						'enableOpenverseMediaCategory',
-						'styles',
-						'template',
-						'templateLock',
-						'titlePlaceholder',
-						'supportsLayout',
-						'widgetTypesToHideFromLegacyWidgetBlock',
-						'__unstableResolvedAssets',
-					].includes( key )
-				)
-			),
+			...settings,
```

Comment:
Could you elaborate on this change? It looks like a great cleanup, but it would mean that the filter with allowed settings was unnecessary. Any idea why it was created in the first place and why it's not necessary now?

Reply:
This and the `supportsLayout` change were merged separately in #47287, there are comments that discuss and explain both decisions.